File: 2023/Solutions/Python/aoc_utils.py
from pathlib import Path
from icecream import ic

BASE = Path(f"C:/syncthing/shared_code_tests/adventOfCode/2023")

def save_path(day: int) -> str:
    """
    Returns the path to save optional files for the given day.
    """
    return BASE / f"day_{day:02}" 

def get_day_input(day: int, test: bool = False, second: bool = False) -> list:
    """
    Returns the input as list of rows for the given day.
    input will be in the day folder (day_XX) and filename are
        input.txt
        test_input.txt
    """
    file_name = f"{"second_test_" if (test and second) else "test_" if test else ""}input.txt"
    file_path = BASE / f"day_{day:02}" / file_name

    #ic(file_path)
    return file_path.read_text().split("\n")

def get_file_path(day: int, test: bool = False):
    """
    Just returns the full path of the file
    """
    file_name = f"{"test_" if test else ""}input.txt"
    file_path = BASE / f"day_{day:02}" / file_name
    return file_path


def read_strip(day: int, test: bool = False) -> list:
    """
    Returns the input as list of rows for the given day.
    input will be in the day folder (day_XX) and filename are
        input.txt
        test_input.txt
    """
    file_path = get_file_path(day, test)

    #ic(file_path)
    return file_path.read_text().strip()

File: 2023/Solutions/perl/aoc_utils.pl
use strict;
use warnings;
use Path::Tiny;

my $BASE = Path::Tiny::path("C:/Users/raikoug/SyncThing/shared_code_tests/adventOfCode/2023/");

sub get_day_input {
    my ($day, $test) = @_;
    $test //= 0;
    my $file_name = ($test ? "test_" : "") . "input.txt";
    my $file_path = $BASE->child(sprintf("day_%02d", $day), $file_name);
    return split "\n", $file_path->slurp;
}

sub get_file_path {
    my ($day, $test) = @_;
    $test //= 0;
    my $file_name = ($test ? "test_" : "") . "input.txt";
    my $file_path = $BASE->child(sprintf("day_%02d", $day), $file_name);
    return $file_path;
}